pkg/apis/kubeone/config/migrate.go

```diff
@@ -17,45 +17,113 @@ limitations under the License.
 package config
 
 import (
-	"fmt"
+	"bytes"
 	"os"
+	"reflect"
 
+	"gopkg.in/yaml.v2"
+
+	kubeoneapi "k8c.io/kubeone/pkg/apis/kubeone"
+	"k8c.io/kubeone/pkg/apis/kubeone/scheme"
 	kubeonev1beta2 "k8c.io/kubeone/pkg/apis/kubeone/v1beta2"
 	kubeonev1beta3 "k8c.io/kubeone/pkg/apis/kubeone/v1beta3"
 	"k8c.io/kubeone/pkg/fail"
 	"k8c.io/kubeone/pkg/yamled"
+
+	kyaml "sigs.k8s.io/yaml"
 )
 
-// MigrateOldConfig migrates KubeOneCluster v1beta1 object to v1beta2
-func MigrateOldConfig(clusterFilePath string) (interface{}, error) {
-	oldConfig, err := loadClusterConfig(clusterFilePath)
+// MigrateV1beta2V1beta3 migrates KubeOneCluster v1beta2 object to v1beta3
+func MigrateV1beta2V1beta3(clusterFilePath string) ([]byte, error) {
```

Comment:
We should add some unit tests for this command.

Reply:
done